services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py

```diff
@@ -469,23 +470,55 @@ async def get_project_services_access_for_gid(request: web.Request) -> web.Respo
         ]
     )
 
+    # Initialize groups to compare with everyone group ID
+    groups_to_compare = {EVERYONE_GROUP_ID}
+
+    # Get the group from the provided group ID
+    _sharing_with_group: Group | None = await get_group_from_gid(
+        app=request.app, gid=query_params.for_gid
+    )
+
+    # Check if the group exists
+    if _sharing_with_group is None:
+        msg = f"Group ID does not exist: {query_params.for_gid}"
+        raise ValueError(msg)
+
+    # Update groups to compare based on the type of sharing group
+    if _sharing_with_group.group_type == GroupTypeInModel.PRIMARY:
+        _user_id = await get_user_id_from_gid(
```

Comment:
THOUGHT: perhaps we should extend the groups-plugin API in a comprehensive way. There as different ways to  uniquely identify a user (sort of primary key), namely: `user_id`, `user_name`, `user_email` or `user_primary_group_id`, therefore any of those should be useable in any of the functions interfaces. 
this way we sould avoid these unecessary calls

Reply:
Not sure whether I fully get what you mean. I am reusing already existing function `get_user_id_from_gid` which is a valid way how to get user ID from his primary group id (side note: generally I would expect to get user resource).

Reply:
I mean, any function in the API for users, should accept any of the unique identifiers for a user. For example
```python
get_user(id=1) == get_user(gid=10) == get_user(name="foo") ==get_user(email="foo@mail.com")
```
should all be possible

Reply:
aha, yes valid point, we should extend it like that :+1: